shared/teams/team/member/edit-channel.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react'
+import * as Kb from '../../../common-adapters'
+import * as Container from '../../../util/container'
+import * as Constants from '../../../constants/teams'
+import * as Styles from '../../../styles'
+import {ModalTitle} from '../../common'
+import * as Types from '../../../constants/types/teams'
+import * as ChatTypes from '../../../constants/types/chat2'
+import * as TeamsGen from '../../../actions/teams-gen'
+import * as RouteTreeGen from '../../../actions/route-tree-gen'
+
+type Props = Container.RouteProps<{
+  channelname: string
+  description: string
+  teamID: Types.TeamID
+  conversationIDKey: ChatTypes.ConversationIDKey
+}>
+
+const EditChannel = (props: Props) => {
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+  const conversationIDKey = Container.getRouteProps(props, 'conversationIDKey', '')
+  const oldName = Container.getRouteProps(props, 'channelname', '')
+  const oldDescription = Container.getRouteProps(props, 'description', '')
+
+  const [name, _setName] = React.useState(oldName)
+  const setName = (newName: string) => _setName(newName.replace(/[^a-zA-Z0-9_]/, ''))
+
+  const [description, setDescription] = React.useState(oldDescription)
+
+  const onBack = () => dispatch(nav.safeNavigateUpPayload())
+  const onClose = () => dispatch(RouteTreeGen.createClearModals())
+
+  const onSave = () => {
```

Comment:
`add-to-channels` needs to reload metas when this is done, on mobile rn you get old data. desktop seems fine